src/nccl_ofi_net.c

```diff
@@ -1543,6 +1580,28 @@ static ncclResult_t ofi_accept(void *listenComm, void **recvComm)
 			goto exit;
 	}
 
+	/* Get local EP address to detect requests to self */
+	ret = fi_getname(&(nccl_ofi_component[dev]->ep->fid),
+	                 (void *)&local_ep_addr,
+	                 &namelen);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Call to fi_getname() failed with RC: %d, ERROR: %s",
+		              ret, fi_strerror(-ret));
+		ret = ncclSystemError;
+		goto error;
+	}
+
+	if (memcmp(local_ep_addr, remote_ep_addr, namelen) == 0) {
+		NCCL_OFI_TRACE(NCCL_NET, "Accept from self");
+		if (request_to_self != NULL) {
+			free_nccl_ofi_req(request_to_self, false);
```

Comment:
Completions processed here.
In case if provider does not implement rendezvous, everything remains the same, fi_trecv() at line 1554 will receive buffer as usual. The only difference is that we do not forcedly wait for completion in ofi_connect().

Reply:
Shouldn't the completions be polled from `fi_cq_read`? We are making an assumption here that the request submitted to self doesn't require completions processing by applications.